OutOfSchool/OutOfSchool.WebApi/Services/CompanyInformationService.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using OutOfSchool.Services.Enums;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.WebApi.Models;
+
+namespace OutOfSchool.WebApi.Services
+{
+    /// <summary>
+    /// Implements the interface with CRUD functionality for the CompanyInformation entities (AboutPortal, SupportInformation and LawsAndRegulations).
+    /// </summary>
+    public class CompanyInformationService : ICompanyInformationService
+    {
+        private const int LimitOfItems = 10;
+
+        private readonly IEntityRepository<CompanyInformation> companyInformationRepository;
+        private readonly ILogger<CompanyInformationService> logger;
+        private readonly IMapper mapper;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CompanyInformationService"/> class.
+        /// </summary>
+        /// <param name="companyInformationRepository">CompanyInformation repository.</param>
+        /// <param name="logger">Logger.</param>
+        /// <param name="mapper">Mapper.</param>
+        public CompanyInformationService(
+            IEntityRepository<CompanyInformation> companyInformationRepository,
+            ILogger<CompanyInformationService> logger,
+            IMapper mapper)
+        {
+            this.companyInformationRepository = companyInformationRepository;
```

Comment:
Please ensure all ctor arguments are correct, you can use `?? throw new ArgumentNullException(...)` 

This has to be done, as through all places in class you are using repo/mapper/logger without null-check so your class does require them to be not-null and it's better to know immediately if they are not.